clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -511,6 +511,13 @@ static void addImpliedArgs(const llvm::Triple &Triple,
       BeArgs.push_back("-g");
   if (Args.getLastArg(options::OPT_O0))
     BeArgs.push_back("-cl-opt-disable");
+  // Check if -ffast-math or -funsafe-math.
+  Arg *A = Args.getLastArg(options::OPT_ffast_math, options::OPT_fno_fast_math,
+                           options::OPT_funsafe_math_optimizations,
+                           options::OPT_fno_unsafe_math_optimizations);
+  if (A && (A->getOption().getID() == options::OPT_ffast_math ||
+            A->getOption().getID() == options::OPT_funsafe_math_optimizations))
```

Comment:
`-ffp-model=fast` enables `-ffast-math` as well, should we cover that too?

Reply:
Done.